_posts/2015-03-31-how-we-do-css.md

```diff
@@ -0,0 +1,173 @@
+
+
+### CSS structure
+This is an example of how we expect our stylesheets to be laid out. This is not an actual file.
+
+<script src="https://gist.github.com/neptunz/6b55d2e0db2d0190c540.js"></script>
+
+
+#### + Each file is named after the page or the page module it supports
+#### + We never style ID’s only Classes
+#### + Keep nesting down to a minimum and no deeper than 4 levels including pseudo selectors
+#### + Group related properties wherever possible
+#### + Media queries go after properties and the order should reflect mobile first
+
+
+### Naming
+
+
+#### Variables and Mixins
+Our variables and mixing naming is pretty specific in order to avoid any confusion. We have been here long enough to understand that the company likes to make simple to drastic changes to support the business in general. We have to develop keeping scalability in mind and this variable/mixin strategy just makes more sense for us.
+
+
+#### Mixin
+
+<script src="https://gist.github.com/neptunz/65b4c6cc19a4c4ad2fd6.js"></script>
+
+
+#### Color Variables
+
+<script src="https://gist.github.com/neptunz/c473d1fb7056fb22ef46.js"></script>
+
+
+#### Classes
+Our preferred technique is a more descriptive naming convention over deep nesting.
+
+<script src="https://gist.github.com/neptunz/2c49168fb55be1cbbd25.js"></script>
+
+
+#### Media Queries
+For the most part we target based on viewport width. We have a feature or two that depends on the detection of touch. We stick to three basic breakpoints and rarely make exceptions for custom breakpoints. We don’t want to compromise code quality to consider every viewport width consideration.
+
+<script src="https://gist.github.com/neptunz/b1467ce5f11c61b22297.js"></script>
+
+
+#### Typography
+
+<script src="https://gist.github.com/neptunz/d2ce9353c6f1025922df.js"></script>
+
+
+#### Linting
+Currently our linting consists of pull request code reviews. This is what we look for:
+
++ Levels of nesting
++ Indentation
++ Flag the use of ID's
++ Spacing after selectors and properties
++ Formatting of properties
++ Code that's commented out
++ Formatting of selectors - lowercase and hyphened preferred
++ Null values should be 0 or none instead of 0px
++ Use of color names should use variables
++ Duplicate selectors or properties
+
+
+#### The fun never ends
+This post and our style guide that is soon to come will never be a done deal written in stone. What we build today is cutting edge and future proof until the future is now and we realize how crazy we were when we developed using archaic methods and tools. The tech world moves at an insane pace and that's what makes our job fun. New technologies will emerge and we will use them to make our applications and our lives as developers better. Therefore, this is relevant for the time being and I hope that it covers us for a very long time, but we all can't be surprised if we have to come back and revise this to support new new. Until then, we will enjoy this sandwich.
```

Comment:
This post and our <strong>forthcoming</strong> style guide <strike>that is soon to come</strike> will never be a done deal written in stone.